MenuApplets/Audio/main.cpp

```diff
@@ -52,21 +58,88 @@ class AudioWidget final : public GUI::Widget {
+        m_root_container->set_fill_with_background_color(true);
+        m_root_container->set_size_policy(GUI::SizePolicy::Fill, GUI::SizePolicy::Fill);
+        m_root_container->set_layout<GUI::VerticalBoxLayout>();
+        m_root_container->layout()->set_margins({ 0, 4, 0, 4 });
+        m_root_container->layout()->set_spacing(0);
+        m_root_container->set_frame_thickness(2);
+        m_root_container->set_frame_shape(Gfx::FrameShape::Container);
+        m_root_container->set_frame_shadow(Gfx::FrameShadow::Raised);
+
+        m_percent_box = m_root_container->add<GUI::CheckBox>("\xE2\x84\xB9");
+        m_percent_box->set_size_policy(GUI::SizePolicy::Fixed, GUI::SizePolicy::Fixed);
+        m_percent_box->set_preferred_size(27, 16);
+        m_percent_box->set_checked(false);
+        m_percent_box->set_tooltip("Show percent");
+        m_percent_box->on_checked = [&](bool show_percent) {
+            m_show_percent = show_percent;
+            if (!m_show_percent) {
+                window()->resize(16, 16);
+                m_percent_box->set_tooltip("Show percent");
+            } else {
+                window()->resize(44, 16);
+                m_percent_box->set_tooltip("Hide percent");
+            }
+            reposition_slider_window();
+            GUI::Application::the()->hide_tooltip();
+        };
+
+        m_slider = m_root_container->add<GUI::VerticalSlider>();
+        m_slider->set_max(20);
+        m_slider->set_value(0);
+        m_slider->set_knob_size_mode(GUI::Slider::KnobSizeMode::Proportional);
+        m_slider->set_size_policy(GUI::SizePolicy::Fill, GUI::SizePolicy::Fill);
+        m_slider->on_value_changed = [&](int value) {
+            int volume = clamp((20 - value) * 5, 0, 100);
+            m_audio_client->set_main_mix_volume(volume);
+            update();
+        };
+
+        m_mute_box = m_root_container->add<GUI::CheckBox>("\xE2\x9D\x8C");
+        m_mute_box->set_size_policy(GUI::SizePolicy::Fixed, GUI::SizePolicy::Fixed);
+        m_mute_box->set_preferred_size(27, 16);
+        m_mute_box->set_checked(false);
+        m_mute_box->set_tooltip("Mute");
+        m_mute_box->on_checked = [&](bool is_muted) {
+            is_muted ? m_mute_box->set_tooltip("Unmute") : m_mute_box->set_tooltip("Mute");
+            m_audio_client->set_muted(is_muted);
+            GUI::Application::the()->hide_tooltip();
+        };
     }
 
-    virtual ~AudioWidget() override {}
+    virtual ~AudioWidget() override { }
 
 private:
     virtual void mousedown_event(GUI::MouseEvent& event) override
     {
-        if (event.button() != GUI::MouseButton::Left)
+        if (event.button() == GUI::MouseButton::Left) {
+            if (!m_slider_window->is_visible())
+                open();
+            else
+                close();
             return;
-        m_audio_client->set_muted(!m_audio_muted);
-        update();
+        }
+        if (event.button() == GUI::MouseButton::Right) {
```

Comment:
You could remove the above early `return` if you added an `else` to this `if`, or perhaps make the whole thing into a `switch` statement

Reply:
True, this is just the style I see most often in mouse/key_events